### ISSUE TITLE
Compilation error in Ubuntu 16.04

### ISSUE THREAD

**Issue by michaelsjackson** [poke1024/godot_vector_graphics#2]
```
[Initial build] Compiling ==> modules/vector_graphics/register_types.cpp
In file included from modules/vector_graphics/vector_graphics_path.h:8:0,
                 from modules/vector_graphics/register_types.cpp:7:
modules/vector_graphics/utils.h:12:37: fatal error: tove2d/src/cpp/graphics.h: No such file or directory
compilation terminated.
scons: *** [modules/vector_graphics/register_types.x11.tools.64.o] Error 1
scons: building terminated because of errors.
```

**Comment by michaelsjackson** [comment]
After copying tove2d files, now I am getting another error:

> modules/vector_graphics/utils.cpp: In function 'tove::PathRef new_transformed_path(const PathRef&, const Transform2D&)':
modules/vector_graphics/utils.cpp:19:16: error: 'class tove::Path' has no member named 'setIndex'
     tove_path->setIndex(p_tove_path->getIndex());
                ^
modules/vector_graphics/utils.cpp:19:38: error: 'class tove::Path' has no member named 'getIndex'
     tove_path->setIndex(p_tove_path->getIndex());
                                      ^
modules/vector_graphics/utils.cpp: In function 'Ref<ShaderMaterial> copy_mesh(Ref<ArrayMesh>&, tove::MeshRef&, const GraphicsRef&, Ref<Texture>&)':
modules/vector_graphics/utils.cpp:43:18: error: 'class tove::AbstractMesh' has no member named 'copyVertexData'
     p_tove_mesh->copyVertexData(vertices, size);
                  ^
modules/vector_graphics/utils.cpp:115:22: error: 'struct ToveGradientData' has no member named 'arguments'
         gradientData.arguments = arguments_data_write.ptr();
                      ^
modules/vector_graphics/utils.cpp:120:15: error: 'class tove::ColorFeed' has no member named 'bind'
         feed->bind(gradientData);
               ^
modules/vector_graphics/utils.cpp:246:12: warning: unused variable 'surface' [-Wunused-variable]
  const int surface = p_mesh->get_surface_count();
            ^
scons: *** [modules/vector_graphics/utils.x11.tools.64.o] Error 1
scons: building terminated because of errors.

**Comment by poke1024** [comment]
I just fixed the compilation for Ubuntu 18.04.

The error above looks like another problem though, namely like you cloned the repo without sub repos (i.e. `git clone`). Have you tried `git clone --recurse-submodules https://github.com/poke1024/godot_vector_graphics`?

The checked out folder should look like this:

```
(inside your_godot/modules/)

vector_graphics/
├── patches
│   └── icons
└── tove2d
    ├── demos
    │   ├── alpha
    │   │   └── assets -> ../assets
    │   ├── assets
    │   │   └── amatic
    │   ├── blob
    │   │   └── assets -> ../assets
    │   ├── clippath
    │   │   └── assets -> ../assets
    │   ├── debug
    │   │   ├── debug-gpux
    │   │   │   └── assets -> ../../assets
    │   │   └── simple
    │   ├── editor
    │   │   ├── assets -> ../assets
    │   │   ├── boxy
    │   │   │   └── cassowary
    │   │   ├── images
    │   │   │   └── thirdparty
    │   │   ├── tools
    │   │   └── ui
    │   ├── fillrule
    │   │   └── assets -> ../assets
    │   ├── gradients
    │   │   └── assets -> ../assets
    │   ├── hearts
    │   │   └── assets -> ../assets
    │   ├── renderers
    │   │   └── assets -> ../assets
    │   ├── tesselation
    │   │   └── assets -> ../assets
    │   └── zoom
    │       └── assets -> ../assets
    ├── docs
    │   └── images
    │       ├── demos
    │       └── tutorial
    └── src
        ├── cpp
        │   ├── gpux
        │   ├── interface
        │   ├── mesh
        │   └── shader
        │       └── feed
        ├── glsl
        ├── lua
        │   └── core
        └── thirdparty
            ├── fp16
            │   ├── bench
            │   ├── cmake
            │   ├── include
            │   │   └── fp16
            │   ├── jni
            │   ├── test
            │   │   └── peachpy
            │   └── third-party
            ├── nanosvg
            │   ├── example
            │   ├── src
            │   └── tove
            ├── polypartition
            │   ├── images
            │   ├── src
            │   └── test
            ├── robin-map
            │   ├── cmake
            │   ├── include
            │   │   └── tsl
            │   └── tests
            └── tinyxml2
                ├── contrib
                ├── docs
                │   └── search
                ├── resources
                │   └── out
                └── tinyxml2
                    ├── tinyxml2-cbp
                    └── tinyxml2.xcodeproj
```

**Comment by michaelsjackson** [comment]
I did, but retrying again.
